autobump-maven/src/main/java/com/github/autobump/maven/model/MavenDependencyBumper.java

```diff
@@ -0,0 +1,62 @@
+package com.github.autobump.maven.model;
+
+import com.github.autobump.core.exceptions.DependencyParserException;
+import com.github.autobump.core.model.Bump;
+import com.github.autobump.core.model.Dependency;
+import com.github.autobump.core.model.DependencyBumper;
+import com.github.autobump.core.model.Workspace;
+import org.apache.maven.model.InputLocation;
+import org.apache.maven.model.InputSource;
+import org.apache.maven.model.io.xpp3.MavenXpp3ReaderEx;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+
+public class MavenDependencyBumper implements DependencyBumper {
+
+    @Override
+    public void bump(Workspace workspace, Bump bump) {
+        try (Reader reader = workspace.getDependencyDocument(MavenDependencyResolver.DEPENDENCY_FILENAME)) {
+            InputLocation versionLocation = findVersionLine(reader, bump.getDependency(), workspace.getProjectRoot());
+            updateDependency(versionLocation, bump);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void updateDependency(InputLocation versionLocation, Bump bump) throws IOException {
+        Path file = Paths.get(versionLocation.getSource().getLocation());
+        List<String> out = Files.readAllLines(file);
+        out.set(versionLocation.getLineNumber() - 1,
+                out.get(versionLocation.getLineNumber() - 1)
+                        .replace(bump.getDependency().getVersion(),
+                                bump.getVersion().getVersionNumber()));
+        Files.write(file, out, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    private InputLocation findVersionLine(Reader reader, Dependency dependency, String rootdir) throws IOException {
+        MavenXpp3ReaderEx mavenXpp3ReaderEx = new MavenXpp3ReaderEx();
+        try {
+            InputSource inputSource = new InputSource();
+            inputSource.setLocation(rootdir + "/pom.xml");
+            return mavenXpp3ReaderEx.read(reader, true, inputSource)
+                    .getDependencies()
+                    .stream()
+                    .filter(dependency1 ->
+                            dependency1.getGroupId().equals(dependency.getGroup())
+                                    && dependency1.getArtifactId().equals(dependency.getName()))
+                    .findFirst()
+                    .orElseThrow()
+                    .getLocation("version");
+        } catch (XmlPullParserException e) {
```

Comment:
like the IOException above you could give this class a different implementation of the MavenXpp3ReaderEx which just throws XmlPullParserException when the read method is called